MiddleClick/Controller.m

```diff
@@ -115,49 +112,109 @@ - (void)start
   // reconifguration of Core Graphics
   CGDisplayRegisterReconfigurationCallback(displayReconfigurationCallBack, self);
   
-  // we only want to see left mouse down and left mouse up, because we only want
-  // to change that one
-  CGEventMask eventMask = (CGEventMaskBit(kCGEventLeftMouseDown) | CGEventMaskBit(kCGEventLeftMouseUp));
-  
-  // create eventTap which listens for core grpahic events with the filter
-  // sepcified above (so left mouse down and up again)
-  CFMachPortRef eventTap = CGEventTapCreate(
-                                            kCGHIDEventTap, kCGHeadInsertEventTap, kCGEventTapOptionDefault,
-                                            eventMask, mouseCallback, NULL);
-  
-  if (eventTap) {
-    // Add to the current run loop.
-    CFRunLoopSourceRef runLoopSource = CFMachPortCreateRunLoopSource(kCFAllocatorDefault, eventTap, 0);
-    CFRunLoopAddSource(CFRunLoopGetCurrent(), runLoopSource,
-                       kCFRunLoopCommonModes);
-    
-    // Enable the event tap.
-    CGEventTapEnable(eventTap, true);
+  [self registerMouseCallback:pool];
```

Comment:
We do
```
  NSAutoreleasePool* pool = [[NSAutoreleasePool alloc] init];
  [NSApplication sharedApplication];
```
and 
```
registerTouchCallback();
```
above, so could we just call `startUnstableListeners` here instead?

Reply:
Seems possible, but I don't want to mess with the autorelease pool. Since I'm not sure how to measure its effect, I won't be aware if it doesn't work correctly.

P.S. Probably broke it already tbh, never seen a pool passed as a method parameter.